controller/konnect/konnectextension_controller.go

```diff
@@ -68,7 +85,16 @@ func (r *KonnectExtensionReconciler) SetupWithManager(ctx context.Context, mgr c
 				),
 			),
 		).
-		// TODO: watch secrets https://github.com/Kong/gateway-operator/issues/1210
+		Watches(
+			&corev1.Secret{},
+			handler.EnqueueRequestsFromMapFunc(
+				konnectExtensionReconcileRequestsForSecret(mgr.GetClient()),
+			),
+			builder.WithPredicates(
+				labelSelectorPredicate,
+				predicate.NewPredicateFuncs(secretUsedByKonnectExtension(mgr.GetClient())),
+			),
```

Comment:
That's essentially what I proposed in 1) above.

```suggestion
			&corev1.Secret{},
			handler.EnqueueRequestsFromMapFunc(
				konnectExtensionReconcileRequestsForSecret(mgr.GetClient()),
			),
			builder.WithPredicates(				predicate.NewPredicateFuncs(secretUsedByKonnectExtension(mgr.GetClient())),
			),
```